src/WebJobs.Script.WebHost/WebScriptHostHttpRoutesManager.cs

```diff
@@ -69,7 +69,7 @@ public void InitializeHttpFunctionRoutes(IScriptJobHost host)
 
                     LogRouteMap(routesLogBuilder, function.Metadata.Name, route, httpTrigger.Methods, isProxy, _httpOptions.Value.RoutePrefix);
                 }
-                else if (warmupRouteBuilder != null && function.IsWarmupFunction())
+                else if (warmupRouteBuilder != null && !_environment.IsInValidationMode() && function.IsWarmupFunction())
```

Comment:
Same here, looks like it will impact non-flex skus. We should ensure this doesn't happen.